opensearch-transport/lib/opensearch/transport/client.rb

```diff
@@ -330,14 +330,17 @@ def __parse_host(host)
       # @api private
       #
       def __auto_detect_adapter
+        # Get the Faraday adapter list without initializing it.
+        adapters = Faraday::Adapter.instance_variable_get(:@registered_middleware) || {}
```

Comment:
This is the fastest reliable way to get the adapter collection that works in Faraday 1 and 2. Faraday 2 has a `registered_middleware` method which retrieves the collection (and initializes the instance variable), but Faraday 1 only has an undocumented `fetch_middleware` method that retrieves from the middleware. Both define the same instance variable though for the collection.